src/Player.js

```diff
@@ -61,17 +61,22 @@ Player.prototype._add = function (playable) {
 	return false;
 };
 
-Player.prototype._remove = function (playable) {
+Player.prototype._remove = function (playable, immediate) {
 	if (playable._handle === null) {
 		this._warn('[Player._remove] Playable is not being used');
 		return false;
 	}
 
 	// Playable is handled, either by this player or by another one
 	if (playable._handle.container === this._activePlayables) {
-		// Playable was active, adding to remove list
-		playable._handle = this._playablesToRemove.add(playable._handle);
-		return true;
+		if (immediate) {
+			playable._handle = this._activePlayables.removeByReference(playable._handle);
+			return true;
```

Comment:
If true in both cases, maybe more elegant to move the return statement to after the `else`?